system-tests/fortran/configuration.ini

```diff
@@ -12,3 +12,6 @@ rules = MissingOnly
 
 [style.missing_pointer_init]
 rules = MissingPointerInit
+
+[style.wrong_kind]
+rules = KindPattern(r'.*_jazz', r'.*_metal')
```

Comment:
I should perhaps have put this comment in the `KindPattern` source instead but I wonder if it might be better to have the 2 arguments be keywords instead of positional?  So you'd have `KindPattern(integer=r'.*_jazz', real=r'.*_metal')`.  Or alternatively have it be a single argument mapping so `KindPattern({'integer': r'.*_jazz', 'real': r'.*_metal'})` - to avoid having to remember which kinds are supported and in which order they need to be.

It might tie into the earlier comment about how the `defaultdict` is used - if a project _only_ wants to enforce a kind for reals then they could specify only that part.  Appreciate the hard part here is actually parsing the configuration, perhaps it's more trouble than it's worth?

Reply:
I think that's a great idea. Requiring the use of keyword arguments will make things a lot clearer. I have done this which required supporting keyword arguments in rule setup.